core/modules/widgets/widget.js

```diff
@@ -751,23 +751,44 @@ Widget.prototype.findFirstDomNode = function() {
 };
 
 /*
-Remove any DOM nodes created by this widget or its children
+Entry into destroy procedure
+*/
+Widget.prototype.destroyChildren = function() {
+	$tw.utils.each(this.children,function(childWidget) {
+		childWidget.destroy();
+	});
+};
+/*
+Legacy entry into destroy procedure
 */
 Widget.prototype.removeChildDomNodes = function() {
-	// If this widget has directly created DOM nodes, delete them and exit. This assumes that any child widgets are contained within the created DOM nodes, which would normally be the case
+	this.destroy();
+};
+/*
+Default destroy
+*/
+Widget.prototype.destroy = function() {
+	// call children to remove their resources
+	this.destroyChildren();
+	// remove our resources
+	this.children = [];
+	this.removeLocalDomNodes();	
+};
+
+/*
+Remove any DOM nodes created by this widget 
+*/
+Widget.prototype.removeLocalDomNodes = function() {
+	// If this widget has directly created DOM nodes, delete them and exit.
 	if(this.domNodes.length > 0) {
 		$tw.utils.each(this.domNodes,function(domNode) {
 			domNode.parentNode.removeChild(domNode);
```

Comment:
Should check here `domNode.parentNode?.removeChild?.(domNode);`

Otherwise `Uncaught TypeError: Cannot read properties of null (reading 'removeChild')`

Reply:
A userside workaround is add `this.domNodes = [];` in `destroy`, so this line won't get executed.